tests/e2e/gc_rwx_deployments.go

```diff
@@ -0,0 +1,331 @@
+		var missingPod *v1.Pod
+		ignoreLabels := make(map[string]string)
+
+		ginkgo.By("CNS_TEST: Running for GC setup")
+		scParameters[svStorageClassName] = storagePolicyName
+		ginkgo.By("Creating a PVC")
+		storageclasspvc, pvclaim, err = createPVCAndStorageClass(client,
+			namespace, nil, scParameters, diskSize, nil, "", false, v1.ReadWriteMany)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclasspvc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating the PVC2 in guest cluster")
+		pvc2 := getPersistentVolumeClaimSpecForRWX(namespace, nil, "", diskSize)
+		pvc2.Spec.AccessModes[0] = v1.ReadWriteMany
+		pvc2.Spec.StorageClassName = &storageclasspvc.Name
+
+		pvc2, err = client.CoreV1().PersistentVolumeClaims(namespace).Create(ctx, pvc2, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Expect claim to provision volume successfully")
+		persistentvolumes, err := fpv.WaitForPVClaimBoundPhase(client,
+			[]*v1.PersistentVolumeClaim{pvclaim, pvc2}, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to provision volume")
+
+		pvc1NameInSV := persistentvolumes[0].Spec.CSI.VolumeHandle
+		gomega.Expect(pvc1NameInSV).NotTo(gomega.BeEmpty())
+
+		pvc2NameInSV := persistentvolumes[1].Spec.CSI.VolumeHandle
+		gomega.Expect(pvc2NameInSV).NotTo(gomega.BeEmpty())
+
+		fcd1IDInCNS := getVolumeIDFromSupervisorCluster(pvc1NameInSV)
+		gomega.Expect(fcd1IDInCNS).NotTo(gomega.BeEmpty())
+
+		fcd2IDInCNS := getVolumeIDFromSupervisorCluster(pvc2NameInSV)
+		gomega.Expect(fcd2IDInCNS).NotTo(gomega.BeEmpty())
+
+		defer func() {
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, pvclaim.Namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(fcd1IDInCNS)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			err = fpv.DeletePersistentVolumeClaim(client, pvc2.Name, pvc2.Namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(fcd2IDInCNS)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			//Add a check to validate CnsVolumeMetadata crd
+			verifyCRDInSupervisorWithWait(ctx, f, pvc1NameInSV, crdCNSVolumeMetadatas, crdVersion, crdGroup, false)
+			verifyCRDInSupervisorWithWait(ctx, f, pvc2NameInSV, crdCNSVolumeMetadatas, crdVersion, crdGroup, false)
+
+		}()
+
+		// Verify using CNS Query API if VolumeID retrieved from PV is present.
+		ginkgo.By(fmt.Sprintf("Invoking QueryCNSVolumeWithResult with VolumeID: %s", fcd1IDInCNS))
+		queryResult, err := e2eVSphere.queryCNSVolumeWithResult(fcd1IDInCNS)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(queryResult.Volumes).ShouldNot(gomega.BeEmpty())
+		ginkgo.By(fmt.Sprintf("volume Name:%s, capacity:%d volumeType:%s health:%s accesspoint: %s",
```

Comment:
you are just dumping the values where is the verification part

Reply:
also use framework.Logf or logger.info for dumping info logs, we use ginkgo.By() to denote steps

Reply:
Added the validation check now.